### PR TITLE
IAA Role [DNM]

### PR DESCRIPTION
**Pull request by Cdey78**
#1802 , Now with _100%_ less failure!

image
                       ^
Ignore the officer, please, he's a removed role as of now

:cl: Cdey
add: Implements IAA
del: Removes detective
wip: IAA should be going under whitelist eventually
/:cl:

### CODE REVIEW

**Comment by Cdey78** [comment]
Correction, there are locker sprites for the IAA

File: code/game/objects/structures/crates_lockers/closets/secure/misc.dm [hunk 1]
```diff
@@ -65,3 +65,39 @@
 	new /obj/item/weapon/storage/box/metalfoam(src)
 	for(var/i in 1 to 3)
 		new /obj/item/weapon/rcd_ammo/large(src)
+
+		/obj/structure/closet/secure_closet/iaa
+	name = "iaa's Closet"
```
Comment (ike709) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: internal affairs agent's closet

File: code/game/objects/structures/crates_lockers/closets/secure/misc.dm [hunk 2]
```diff
@@ -65,3 +65,39 @@
...
+	new /obj/item/(src)
+	new /obj/item/weapon/clipboard(src)
+	new /obj/item/weapon/melee/classic_baton/telescopic(src)
+	new /obj/item/weapon/restraints/handcuffs/cable/zipties(src)
+	new /obj/item/clothing/under/rank/iaa(src)
+	new /obj/item/clothing/suit/iaa_suit(src)
+	new /obj/item/clothing/shoes/laceup(src)
+
+	/obj/structure/closet/secure_closet/rshield
+	name = "redshield's Closet"
```
Comment (ike709) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: redshield's closet

File: code/game/objects/structures/crates_lockers/closets/secure/misc.dm [hunk 3]
```diff
@@ -65,3 +65,39 @@
 	new /obj/item/weapon/storage/box/metalfoam(src)
 	for(var/i in 1 to 3)
 		new /obj/item/weapon/rcd_ammo/large(src)
+
+		/obj/structure/closet/secure_closet/iaa
+	name = "iaa's Closet"
+	desc = "Internal Affairs Closet"
```
Comment (ike709) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: Surely you can come up with a better description, like the Redshield's.

File: code/game/objects/structures/crates_lockers/closets/secure/misc.dm [hunk 4]
```diff
@@ -65,3 +65,39 @@
...
+	req_access = list(ACCESS_CENT_SPECOPS)
+	icon_state = "iaa"
+
+/obj/structure/closet/secure_closet/iaa/PopulateContents()
+	..()
+	new /obj/item/weapon/storage/belt/holster/full(src)
+	new /obj/item/(src)
```
Comment (ike709) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: what

File: code/game/objects/structures/crates_lockers/closets/secure/misc.dm [hunk 5]
```diff
@@ -65,3 +65,39 @@
...
+	new /obj/item/weapon/clipboard(src)
```
Comment (ike709) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: You should include a briefcase too. And a folder or two.

Comment (Cdey78) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: he should have a folder on him

File: code/game/objects/structures/crates_lockers/closets/secure/misc.dm (hunk 4 above)
Comment (Cdey78) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: oh, whoops

File: code/game/objects/structures/crates_lockers/closets/secure/misc.dm (hunk 5 above)
Comment (Cdey78) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: also, I presume his office should have a folder or two

File: code/modules/jobs/job_types/captain.dm [hunk 6]
```diff
@@ -96,3 +96,43 @@ Executive Officer
...
+/datum/outfit/job/iaa
+	 name = "Internal Affairs Agent"
+		jobtype = /datum/job/iaa
+
+		belt = /obj/item/device/pda/detective
+		ears = /obj/item/device/radio/headset/heads/hos
+		uniform = /obj/item/clothing/under/rank/iaa
+		shoes = /obj/item/clothing/shoes/laceup
+		suit = /obj/item/clothing/suit/iaa_suit
+		l_pocket = /obj/item/weapon/folder/blue
```
Comment (ike709) [inline code/modules/jobs/job_types/captain.dm]: This can be included in the IAA's room.

File: code/modules/jobs/job_types/security.dm [hunk 7]
```diff
@@ -106,55 +106,48 @@ Warden
...
 */
-/datum/job/detective
-	title = "Detective"
-	flag = DETECTIVE
-	department_head = list("Head of Security")
+
+/datum/job/rshield
+	title = "Redshield Officer"
+	flag = RSHIELD
+	department_head = list("Centcomm Official")
```
Comment (ike709) [inline code/modules/jobs/job_types/security.dm]: Central Command

File: code/modules/jobs/job_types/captain.dm [hunk 8]
```diff
@@ -96,3 +96,43 @@ Executive Officer
...
 		/obj/item/weapon/melee/classic_baton/telescopic=1, /obj/item/device/modular_computer/tablet/preset/advanced = 1)
+
+/*
+Internal Affairs Agent
+*/
+
+/datum/job/iaa
+		title = "Internal Affairs Agent"
+		flag = IAA
+		department_head = list("Centcomm Official")
```
Comment (ike709) [inline code/modules/jobs/job_types/captain.dm]: Central Command

File: code/modules/jobs/job_types/security.dm [hunk 9]
```diff
@@ -106,55 +106,48 @@ Warden
...
 	department_flag = ENGSEC
 	faction = "Station"
 	total_positions = 1
 	spawn_positions = 1
-	supervisors = "the head of security"
+	supervisors = "internal affairs agent and the Magistrate"
```
Comment (ike709) [inline code/modules/jobs/job_types/security.dm]: the internal affairs agent

You can alter this when the magistrate is coded.

File: code/game/objects/structures/crates_lockers/closets/secure/misc.dm [hunk 10]
```diff
@@ -65,3 +65,39 @@
 	new /obj/item/weapon/storage/box/metalfoam(src)
 	for(var/i in 1 to 3)
 		new /obj/item/weapon/rcd_ammo/large(src)
+
+		/obj/structure/closet/secure_closet/iaa
```
Comment (ike709) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: This shouldn't be indented.

File: code/game/objects/structures/crates_lockers/closets/secure/misc.dm [hunk 11]
```diff
@@ -65,3 +65,39 @@
...
+	new /obj/item/weapon/storage/belt/holster/full(src)
+	new /obj/item/(src)
+	new /obj/item/weapon/clipboard(src)
+	new /obj/item/weapon/melee/classic_baton/telescopic(src)
+	new /obj/item/weapon/restraints/handcuffs/cable/zipties(src)
+	new /obj/item/clothing/under/rank/iaa(src)
+	new /obj/item/clothing/suit/iaa_suit(src)
+	new /obj/item/clothing/shoes/laceup(src)
+
+	/obj/structure/closet/secure_closet/rshield
```
Comment (ike709) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: This shouldn't be indented.

File: code/modules/clothing/under/miscellaneous.dm [hunk 12]
```diff
@@ -728,3 +728,12 @@
 	icon_state = "hostanclothes"
 	item_state = "hostanclothes"
 	item_color = "hostanclothes"
+
+	/obj/item/clothing/under/rank/iaa
```
Comment (ike709) [inline code/modules/clothing/under/miscellaneous.dm]: This shouldn't be indented.

File: code/game/objects/structures/crates_lockers/closets/secure/misc.dm (hunk 5 above)
Comment (Cdey78) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: Alright

File: code/modules/jobs/jobs.dm [hunk 13]
```diff
@@ -52,8 +53,9 @@ GLOBAL_LIST_INIT(civilian_positions, list(
 GLOBAL_LIST_INIT(security_positions, list(
 	"Head of Security",
 	"Master-at-Arms",
+	"Redshield Officer",
 	"Detective",
```
Comment (ike709) [inline code/modules/jobs/jobs.dm]: Remove "Detective",

File: code/__DEFINES/jobs.dm [hunk 14]
```diff
@@ -13,6 +13,8 @@
 #define AI_JF				(1<<9)
```
Comment (ike709) [inline code/__DEFINES/jobs.dm]: I can't comment on the correct line but you need to remove

#define DETECTIVE			(1<<3)

You can put redshield in its place and change the number from 13 to 3

File: code/game/objects/structures/crates_lockers/closets/secure/misc.dm (hunk 1 above)
Comment (Qustinnus) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: atleast make it capitalized if you make acronyms in the future.

**Comment by Qustinnus** [comment]
redshield is dumb as hell, why do we need a snowflake whitelist job that's basically a security officer on steroids that gets to act like an even bigger dick because "muh centcom".

Remove redshields and keep IAA because IAA actually is useful and interesting. Redshield is just snowflakey and dumb to have as a roundstart job.

File: code/game/objects/structures/crates_lockers/closets/secure/misc.dm (hunk 1 above)
Comment (ike709) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: names are entirely lowercase floyd

File: code/game/objects/structures/crates_lockers/closets/secure/misc.dm [hunk 15]
```diff
@@ -65,3 +65,39 @@
...
+
+		/obj/structure/closet/secure_closet/iaa
+	name = "iaa's Closet"
+	desc = "Internal Affairs Closet"
+	req_access = list(ACCESS_CENT_SPECOPS)
+	icon_state = "iaa"
+
+/obj/structure/closet/secure_closet/iaa/PopulateContents()
+	..()
+	new /obj/item/weapon/storage/belt/holster/full(src)
```
Comment (ike709) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: It was discussed... Last night I think? Maybe the day before? And ecolli has been convinced that the IAA does not need a gun. Remove the thing.

**Comment by Cdey78** [comment]
Ecolli I believe asked for the redshield, FYI..

**Comment by ike709** [comment]
But he did agree that the IAA does not need a gun.

**Comment by Cdey78** [comment]
I know, I already removed it from the locker.

**Comment by Qustinnus** [comment]
i veto redshields with my bi-weekly veto coin.

**Comment by Cdey78** [comment]
I'll comment it out until Ecolli confirms its removal

**Comment by Cdey78** [comment]
FYI Redshields job would have no authority over anyone, to be honest

**Comment by Cdey78** [comment]
also, he doesn't even get that mutch of an equipment upgrade..

**Comment by Qustinnus** [comment]
then there's no point to have him and even then people will assume they are the boss when you put a centcom sticker on them.

**Comment by Cdey78** [comment]
If he's a roundstart role, he's not centcomm, he's a form of ship security that ensures safety of guests

**Comment by ike709** [comment]
Just make a damn Redshield admin button like ERT

**Comment by Cdey78** [comment]
I never said I wasn't going to do that..

**Comment by ike709** [comment]
Or move it to a separate PR so IAA gets merged

**Comment by ike709** [comment]
Separate PR is probably a better idea. Everybody is happy with IAA how it is.

File: code/__DEFINES/jobs.dm [hunk 16]
```diff
@@ -13,6 +13,7 @@
 #define AI_JF				(1<<9)
 #define CYBORG			(1<<10)
 #define BOFFICER  (1<<11)
+//#define RSHIELD  (1<<13)
```
Comment (ike709) [inline code/__DEFINES/jobs.dm]: 12 not 13

**Comment by Cdey78** [comment]
Should be merge ready now?

**Comment by Cdey78** [comment]
nevermind

File: code/modules/jobs/jobs.dm [hunk 17]
```diff
@@ -52,8 +53,8 @@ GLOBAL_LIST_INIT(civilian_positions, list(
 GLOBAL_LIST_INIT(security_positions, list(
 	"Head of Security",
 	"Master-at-Arms",
-	"Detective",
-	"Security Officer"))
+	//"Redshield Officer",
+	"Security Officer")) //MAGISTRATE
```
Comment (ike709) [inline code/modules/jobs/jobs.dm]: Either

//TODO: Magistrate

Or just remove the comment. TODO is a special snowflake thing.

Comment (Cdey78) [inline code/modules/jobs/jobs.dm]: okden

**Comment by ike709** [comment]
You still have to fix those indents too on some lines.

**Comment by Cdey78** [comment]
Thought I did..

**Comment by ike709** [comment]
Also I doubt ninja will merge a PR that adds a commented out job, but ¯\_(ツ)_/¯

**Comment by Cdey78** [comment]
We'll see

**Comment by Cdey78** [comment]
Anything else?

File: code/__DEFINES/jobs.dm [hunk 18]
```diff
@@ -4,7 +4,7 @@
 #define CAPTAIN			(1<<0)
 #define HOS				(1<<1)
 #define MASTERATARMS			(1<<2)
-#define DETECTIVE			(1<<3)
+#define IAA (1<<3)
```
Comment (ike709) [inline code/__DEFINES/jobs.dm]: you fucked up the indention

File: code/game/objects/structures/crates_lockers/closets/secure/misc.dm [hunk 19]
```diff
@@ -65,3 +65,37 @@
 	new /obj/item/weapon/storage/box/metalfoam(src)
 	for(var/i in 1 to 3)
 		new /obj/item/weapon/rcd_ammo/large(src)
+
+		/obj/structure/closet/secure_closet/iaa
```
Comment (ike709) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: this still shouldn't be indented

File: code/game/objects/structures/crates_lockers/closets/secure/misc.dm [hunk 20]
```diff
@@ -65,3 +65,37 @@
...
+/obj/structure/closet/secure_closet/iaa/PopulateContents()
+	..()
+	new /obj/item/weapon/melee/classic_baton/telescopic(src)
+	new /obj/item/weapon/restraints/handcuffs/cable/zipties(src)
+	new /obj/item/clothing/under/rank/iaa(src)
+	new /obj/item/clothing/suit/iaa_suit(src)
+	new /obj/item/clothing/shoes/laceup(src)
+	new /obj/item/weapon/storage/briefcase/lawyer(src)
+
+	/obj/structure/closet/secure_closet/rshield
```
Comment (ike709) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: This still shouldn't be indented

File: code/game/objects/structures/crates_lockers/closets/secure/misc.dm (hunk 19 above)
Comment (Cdey78) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: I thought  unindented that..

File: code/modules/jobs/jobs.dm [hunk 21]
```diff
@@ -52,8 +53,8 @@ GLOBAL_LIST_INIT(civilian_positions, list(
 GLOBAL_LIST_INIT(security_positions, list(
 	"Head of Security",
 	"Master-at-Arms",
-	"Detective",
-	"Security Officer"))
+	//"Redshield Officer",
+	"Security Officer")) //TO DO: MAGISTRATE
```
Comment (ike709) [inline code/modules/jobs/jobs.dm]: When I said TODO, I meant TODO. Not TO DO. TODO is a special thing.

Comment (Cdey78) [inline code/modules/jobs/jobs.dm]: is that all?

File: code/game/objects/structures/crates_lockers/closets/secure/misc.dm [hunk 22]
```diff
@@ -65,3 +65,37 @@
 	new /obj/item/weapon/storage/box/metalfoam(src)
 	for(var/i in 1 to 3)
 		new /obj/item/weapon/rcd_ammo/large(src)
+
+	/obj/structure/closet/secure_closet/iaa
```
Comment (ike709) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: Still needs to be not indented.

Comment (ike709) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: You fixed the other one at least.

Comment (Cdey78) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: I just unindented that?
What

Comment (Cdey78) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: here

File: code/modules/jobs/job_types/captain.dm [hunk 23]
```diff
@@ -96,3 +96,42 @@ Executive Officer
 	head = /obj/item/clothing/head/xocap
 	backpack_contents = list(/obj/item/weapon/storage/box/ids=1,\
 		/obj/item/weapon/melee/classic_baton/telescopic=1, /obj/item/device/modular_computer/tablet/preset/advanced = 1)
+
+/*
+Internal Affairs Agent
+*/
+
+/datum/job/iaa
+		title = "Internal Affairs Agent"
```
Comment (ike709) [inline code/modules/jobs/job_types/captain.dm]: All these vars for iaa are indented too far by one tab

File: code/modules/jobs/job_types/captain.dm [hunk 24]
```diff
@@ -96,3 +96,42 @@ Executive Officer
...
+		access = list(ACCESS_SEC_DOORS, ACCESS_FORENSICS_LOCKERS, ACCESS_MORGUE, ACCESS_MAINT_TUNNELS, ACCESS_COURT, ACCESS_BRIG, ACCESS_WEAPONS, ACCESS_ALL_PERSONAL_LOCKERS,
+					           ACCESS_RESEARCH, ACCESS_ENGINE, ACCESS_MINING, ACCESS_MEDICAL, ACCESS_CONSTRUCTION, ACCESS_MAILSORTING, ACCESS_CENT_GENERAL,
+					           ACCESS_HEADS, ACCESS_RC_ANNOUNCE, ACCESS_KEYCARD_AUTH, ACCESS_GATEWAY, ACCESS_MAINT_TUNNELS, ACCESS_CENT_SPECOPS)
+		minimal_access = list(ACCESS_SEC_DOORS, ACCESS_FORENSICS_LOCKERS, ACCESS_MORGUE, ACCESS_MAINT_TUNNELS, ACCESS_COURT, ACCESS_BRIG, ACCESS_WEAPONS, ACCESS_ALL_PERSONAL_LOCKERS,
+					           ACCESS_RESEARCH, ACCESS_ENGINE, ACCESS_MINING, ACCESS_MEDICAL, ACCESS_CONSTRUCTION, ACCESS_MAILSORTING,
+					           ACCESS_HEADS, ACCESS_RC_ANNOUNCE, ACCESS_KEYCARD_AUTH, ACCESS_GATEWAY, ACCESS_MAINT_TUNNELS, ACCESS_CENT_SPECOPS)
+
+/datum/outfit/job/iaa
+	 name = "Internal Affairs Agent"
+		jobtype = /datum/job/iaa
```
Comment (ike709) [inline code/modules/jobs/job_types/captain.dm]: Same here. They should line up with name.

**Comment by Cdey78** [comment]
That should be it of the should be its?

File: code/modules/jobs/job_types/captain.dm [hunk 25]
```diff
@@ -96,3 +96,42 @@ Executive Officer
 	head = /obj/item/clothing/head/xocap
 	backpack_contents = list(/obj/item/weapon/storage/box/ids=1,\
 		/obj/item/weapon/melee/classic_baton/telescopic=1, /obj/item/device/modular_computer/tablet/preset/advanced = 1)
+
+/*
+Internal Affairs Agent
+*/
+
+/datum/job/iaa
+	 title = "Internal Affairs Agent"
```
Comment (ike709) [inline code/modules/jobs/job_types/captain.dm]: The iaa vars are all indented with both a space and a tab. It needs to just be a tab.

File: code/modules/jobs/job_types/captain.dm [hunk 26]
```diff
@@ -96,3 +96,42 @@ Executive Officer
...
+
+	 access = list(ACCESS_SEC_DOORS, ACCESS_FORENSICS_LOCKERS, ACCESS_MORGUE, ACCESS_MAINT_TUNNELS, ACCESS_COURT, ACCESS_BRIG, ACCESS_WEAPONS, ACCESS_ALL_PERSONAL_LOCKERS,
+					           ACCESS_RESEARCH, ACCESS_ENGINE, ACCESS_MINING, ACCESS_MEDICAL, ACCESS_CONSTRUCTION, ACCESS_MAILSORTING, ACCESS_CENT_GENERAL,
+					           ACCESS_HEADS, ACCESS_RC_ANNOUNCE, ACCESS_KEYCARD_AUTH, ACCESS_GATEWAY, ACCESS_MAINT_TUNNELS, ACCESS_CENT_SPECOPS)
+	 minimal_access = list(ACCESS_SEC_DOORS, ACCESS_FORENSICS_LOCKERS, ACCESS_MORGUE, ACCESS_MAINT_TUNNELS, ACCESS_COURT, ACCESS_BRIG, ACCESS_WEAPONS, ACCESS_ALL_PERSONAL_LOCKERS,
+					           ACCESS_RESEARCH, ACCESS_ENGINE, ACCESS_MINING, ACCESS_MEDICAL, ACCESS_CONSTRUCTION, ACCESS_MAILSORTING,
+					           ACCESS_HEADS, ACCESS_RC_ANNOUNCE, ACCESS_KEYCARD_AUTH, ACCESS_GATEWAY, ACCESS_MAINT_TUNNELS, ACCESS_CENT_SPECOPS)
+
+/datum/outfit/job/iaa
+ name = "Internal Affairs Agent"
```
Comment (ike709) [inline code/modules/jobs/job_types/captain.dm]: Everything except the name var is fine

**Comment by ike709** [comment]
It's not nearly as difficult as you're making it.

File: code/modules/clothing/under/miscellaneous.dm [hunk 27]
```diff
@@ -728,3 +728,12 @@
 	icon_state = "hostanclothes"
 	item_state = "hostanclothes"
 	item_color = "hostanclothes"
+
+/obj/item/clothing/under/rank/iaa
+	name = "internal affairs uniform"
+	desc = "A uniform tailored just for your local internal affairs agent!"
+	icon_state = "iaa"
```
Comment (ike709) [inline code/modules/clothing/under/miscellaneous.dm]: Didn't we have a whole discussion about how the IAA icons exist in the code already?

Comment (Cdey78) [inline code/modules/clothing/under/miscellaneous.dm]: it uses that icon, that's just for the name and description

Comment (ike709) [inline code/modules/clothing/under/miscellaneous.dm]: why did you change the icon name

Comment (ike709) [inline code/modules/clothing/under/miscellaneous.dm]: Also I checked the DMI. It's not the same icon.

File: code/modules/jobs/job_types/captain.dm [hunk 28]
```diff
@@ -96,3 +96,42 @@ Executive Officer
...
+	jobtype = /datum/job/iaa
+
+	belt = /obj/item/device/pda/detective
+	ears = /obj/item/device/radio/headset/heads/hos
+	uniform = /obj/item/clothing/under/rank/iaa
```
Comment (ike709) [inline code/modules/jobs/job_types/captain.dm]: /obj/item/clothing/under/lawyer/blacksuit

Comment (ike709) [inline code/modules/jobs/job_types/captain.dm]: Make it a subtype and rename it if you *really have to*

File: code/modules/jobs/job_types/captain.dm [hunk 29]
```diff
@@ -96,3 +96,42 @@ Executive Officer
...
+	shoes = /obj/item/clothing/shoes/laceup
+	suit = /obj/item/clothing/suit/iaa_suit
```
Comment (ike709) [inline code/modules/jobs/job_types/captain.dm]: /obj/item/clothing/suit/toggle/lawyer/black

Comment (ike709) [inline code/modules/jobs/job_types/captain.dm]: Same thing. Subtype with it's own name and desc if you have to, but you really don't need to.

File: code/modules/clothing/suits/miscellaneous.dm [hunk 30]
```diff
@@ -545,3 +545,10 @@
 	icon_state = "bedsheet"
 	user_vars_to_edit = list("name" = "Spooky Ghost", "real_name" = "Spooky Ghost" , "incorporeal_move" = INCORPOREAL_MOVE_BASIC, "appearance_flags" = KEEP_TOGETHER|TILE_BOUND, "alpha" = 150)
 	alternate_worn_layer = ABOVE_BODY_FRONT_LAYER //so the bedsheet goes over everything but fire
+
+/obj/item/clothing/suit/iaa_suit
```
Comment (ike709) [inline code/modules/clothing/suits/miscellaneous.dm]: remove this and look at the other comments

File: code/modules/clothing/under/miscellaneous.dm [hunk 31]
```diff
@@ -728,3 +728,12 @@
 	icon_state = "hostanclothes"
 	item_state = "hostanclothes"
 	item_color = "hostanclothes"
+
+/obj/item/clothing/under/rank/iaa
```
Comment (ike709) [inline code/modules/clothing/under/miscellaneous.dm]: remove this and look at the other comments

Comment (Cdey78) [inline code/modules/clothing/under/miscellaneous.dm]: the under suit Is a custom sprite, FYI
it doesn't copy another sprite

File: code/modules/jobs/job_types/captain.dm (hunk 26 above)
Comment (Cdey78) [inline code/modules/jobs/job_types/captain.dm]: removing the space makes it error

File: code/modules/clothing/under/miscellaneous.dm (hunk 31 above)
Comment (ike709) [inline code/modules/clothing/under/miscellaneous.dm]: Didn't we have a whole discussion about how the IAA icons exist in the code already?

Comment (ike709) [inline code/modules/clothing/under/miscellaneous.dm]: Also it looks like a worse version of the existing icon

Comment (ike709) [inline code/modules/clothing/under/miscellaneous.dm]: https://nomemeshere.gyazo.com/7d58771b88f23702ebf849c37e98f9f0

versus

https://nomemeshere.gyazo.com/43c0e85218d5e6d2571978edcb113610

Comment (ike709) [inline code/modules/clothing/under/miscellaneous.dm]: I'll do a discord poll for the two icons if you really want to be annoying about it

File: code/game/objects/items/devices/PDA/PDA_types.dm [hunk 32]
```diff
@@ -56,10 +56,10 @@
 	default_cartridge = /obj/item/weapon/cartridge/security
 	icon_state = "pda-security"
 
-/obj/item/device/pda/detective
-	name = "detective PDA"
+/obj/item/device/pda/iaa
+	name = "iaa PDA"
 	default_cartridge = /obj/item/weapon/cartridge/detective
-	icon_state = "pda-detective"
+	icon_state = "pda-iaa"
```
Comment (ike709) [inline code/game/objects/items/devices/PDA/PDA_types.dm]: You need to rename the icon.

File: code/modules/clothing/suits/miscellaneous.dm [hunk 33]
```diff
@@ -545,3 +545,8 @@
 	icon_state = "bedsheet"
 	user_vars_to_edit = list("name" = "Spooky Ghost", "real_name" = "Spooky Ghost" , "incorporeal_move" = INCORPOREAL_MOVE_BASIC, "appearance_flags" = KEEP_TOGETHER|TILE_BOUND, "alpha" = 150)
 	alternate_worn_layer = ABOVE_BODY_FRONT_LAYER //so the bedsheet goes over everything but fire
+
+/obj/item/clothing/suit/toggle/lawer/black/iaa
```
Comment (ike709) [inline code/modules/clothing/suits/miscellaneous.dm]: As I told you before, this should be in code/modules/clothing/suits/jobs.dm between the lawyer suits and the mime suspenders.

However, the code itself is correct. Just not the location.

File: code/modules/clothing/under/miscellaneous.dm [hunk 34]
```diff
@@ -728,3 +728,8 @@
 	icon_state = "hostanclothes"
 	item_state = "hostanclothes"
 	item_color = "hostanclothes"
+
+/obj/item/clothing/under/lawer/blacksuit/iaa
```
Comment (ike709) [inline code/modules/clothing/under/miscellaneous.dm]: As I told you before, this should be in code/modules/clothing/under/jobs/security.dm

Same as before. Correct code, wrong location.

**Comment by ike709** [comment]
Never ever do any map changes in the same PR as code changes. You're pretty much asking for merge conflicts.

Also, the IAA job isn't actually whitelisted in the code in any way. It's just a normal job.

File: code/__DEFINES/access.dm [hunk 35]
```diff
@@ -65,13 +65,15 @@
...
 
+
 //FTL13
 #define ACCESS_HELM 69
 #define ACCESS_MOFFICE 70
 #define ACCESS_AI_CHAMBER 71
 #define ACCESS_WEAPONS_CONSOLE 72
 #define ACCESS_HELMS_CONSOLE 73
 #define ACCESS_MUNITIONS 74
+#define ACCESS_MAA 75 //MAA's office
```
Comment (ike709) [inline code/__DEFINES/access.dm]: HoS and MAA need this

**Comment by Cdey78** [comment]
I know it's a normal job currently, but someone said they wanted whitelist setup before the PR was merged.

**Comment by Cdey78** [comment]
Was that all?

**Comment by Cdey78** [comment]
Wait, does the access automatically get added to the ID computer, or does it need added manually?

File: code/modules/clothing/suits/miscellaneous.dm (hunk 33 above)
Comment (ike709) [inline code/modules/clothing/suits/miscellaneous.dm]: You forgot to remove this when you moved it.

Comment (Cdey78) [inline code/modules/clothing/suits/miscellaneous.dm]: fuck

Comment (Cdey78) [inline code/modules/clothing/suits/miscellaneous.dm]: Fixed

**Comment by ike709** [comment]
You do need to add ACCESS_MAA to code/modules/jobs/access.dm in /proc/get_all_accesses()

**Comment by ike709** [comment]
Besides the whole "TG code doesn't support job whitelisting at all", this seems good to go.

**Comment by Cdey78** [comment]
okden, time to wait until whitelist is added..

**Comment by Ecolli** [comment]
Current status should be: 
-Redshields are an IAA request only, not roundstart.
-IAA gets a baton for self defense. They are not meant to ENFORCE rules, simply report suggested "job bans" to CC/admins for INVESTIGATION. 
-If we can get whitelists working, I strongly suggest this be a HRP whitelist only job as it needs to be done by players who can RP and know just about every job and how they should be performed.

Remember, this entire role is being created to encourage good RP and weed out shit players who FTL the ship into Syndy capital with no RP reason to do so, etc.

**Comment by Qustinnus** [comment]
and if i see too much cancer of "i am du law" from people that got white listed they're also losing their retard stick so they stop implying their soft power with it.

**Comment by Cdey78** [comment]
Eh, I say let's add it without whitelist for now

**Comment by ike709** [comment]
Map conflicts. I CALLED IT.

**Comment by Cdey78** [comment]
fookin' merge conflicts

**Comment by Cdey78** [comment]
Is it possible to just remove the conflicting file?

**Comment by Cdey78** [comment]
I think I might have fixed it; let's see

**Comment by Cdey78** [comment]
should be merge ready?

**Comment by ike709** [comment]
Looks fine to me.

File: code/__DEFINES/jobs.dm [hunk 36]
```diff
@@ -13,6 +13,7 @@
 #define AI_JF				(1<<9)
 #define CYBORG			(1<<10)
 #define BOFFICER  (1<<11)
+//#define RSHIELD  (1<<12)
```
Comment (Qustinnus) [inline code/__DEFINES/jobs.dm]: remove

File: code/game/objects/structures/crates_lockers/closets/secure/misc.dm [hunk 37]
```diff
@@ -65,3 +65,37 @@
...
+/obj/structure/closet/secure_closet/iaa/PopulateContents()
+	..()
+	new /obj/item/weapon/melee/classic_baton/telescopic(src)
+	new /obj/item/weapon/restraints/handcuffs/cable/zipties(src)
+	new /obj/item/clothing/under/lawer/blacksuit/iaa(src)
+	new /obj/item/clothing/suit/toggle/lawer/black/iaa(src)
+	new /obj/item/clothing/shoes/laceup(src)
+	new /obj/item/weapon/storage/briefcase/lawyer(src)
+
+/obj/structure/closet/secure_closet/rshield
```
Comment (Qustinnus) [inline code/game/objects/structures/crates_lockers/closets/secure/misc.dm]: no need for this rshield is not being added like this.

File: code/modules/jobs/job_types/security.dm [hunk 38]
```diff
@@ -106,55 +106,48 @@ Warden
...
+	department_head = list("Centcomm Official")
 	department_flag = ENGSEC
 	faction = "Station"
 	total_positions = 1
 	spawn_positions = 1
-	supervisors = "the head of security"
+	supervisors = "internal affairs agent"
 	selection_color = "#ffeeee"
-	minimal_player_age = 7
+	minimal_player_age = 10 //Whitelist
```
Comment (Qustinnus) [inline code/modules/jobs/job_types/security.dm]: no need this isn't a roundstart job.

**Comment by Super3222** [comment]
What does Redshield do again?

On a side note can you add more detail into your changelog. ex. explain what the jobs do (and possibly what they get)

**Comment by ike709** [comment]
Redshield was removed.
